odoo/addons/base/models/res_partner.py

```diff
@@ -569,6 +571,14 @@ def create(self, vals_list):
             partner._handle_first_contact_creation()
         return partners
 
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_internal_user(self):
+        for partner in self.exists():
```

Comment:
See the conversation above, `partner` in `self` may have already been deleted, so it could not even have an `active` field. Previously I had this condition : `if self.env['res.partner'].sudo().search([('id', '=', partner.id)], limit=1, count=True)` but kos94ok-3D proposed that i use `self.exists()`.

Without such a protection, one test : `odoo/odoo/addons/base/tests/test_orm.py` will fail (`# Deleting an already deleted record should be simply ignored`)

Reply:
Why "have already been deleted" ? How do you reproduce it ? Purpose of this decorator is to be called at unlink. If record is already unlinked there is an issue to fix elsewhere ...

Reply:
This is how it is reproduced in the test file. I also think this is weird to allow to unlink an already unlinked item, but this case has a test specifically written for it, so maybe it is important. I admit I do not understand why.
```python
p1 = self.env['res.partner'].create({'name': 'W'})
[...]
p1.unlink()
[...]
# Deleting an already deleted record should be simply ignored
self.assertTrue(p1.unlink(), "Re-deleting should be a no-op")
```
this is the content of the test file : https://github.com/odoo/odoo/blob/master/odoo/addons/base/tests/test_orm.py

Reply:
That's strange. Indeed your code should not be adapted due to a test that tries to unlink several times the same record. Otherwise we would have to update all api.ondelete methods to do the same kind of things. The issue here is that exists adds a query that is completely useless.

I would update the test to use another model that is not overrided like partner. Doing low level tests on partner model is always risky because you may conflict with functional requirements. Maybe check to remove a res.country for example or a base model like that in that test.